src/components/Figure/Figure.tsx

```diff
@@ -37,9 +43,33 @@ const Figure: React.FC<FigureProps> = ({
     setLoaded(false);
   }, [image]);
 
+  useEffect(() => {
+    const resizeObserver = new ResizeObserver(
+      (entries: ResizeObserverEntry[]) => {
+        for (let entry of entries) {
+          if (entry.contentBoxSize) {
+            const contentBoxSize: ResizeObserverSize = Array.isArray(
+              entry.contentBoxSize
+            )
+              ? entry.contentBoxSize[0]
+              : entry.contentBoxSize;
+            dispatch({
+              type: "updateItemHeight",
+              itemHeight: contentBoxSize.inlineSize,
+            });
+          }
+        }
+      }
+    );
```

Comment:
Thanks @adamjarling. This was something new for me and I think it might make sense to work this into a hook once I wrap my mind around it a bit more.